src/zeroD/IdealGasConstPressureMoleReactor.cpp

```diff
@@ -79,6 +80,15 @@ void IdealGasConstPressureMoleReactor::eval(double time, double* LHS, double* RH
     // external heat transfer
     mcpdTdt += m_Qdot;
 
+    if (auto* plasma = dynamic_cast<PlasmaPhase*>(m_thermo)) {
+        const double qJ = plasma->jouleHeatingPower(); // σE^2 [W/m^3]
+        const double qElastic = plasma->elasticPowerLoss(); // elastic transfer [W/m^3]
+        const double q_total = (qJ + qElastic) * m_vol; // total power [W]
+        if (std::isfinite(q_total)) {
+            mcpdTdt += q_total;
+        }
+    }
+
```

Comment:
This repeated block is begging to be pulled out into a `protected` function of the `Reactor` class that provides `q_total` so the addition to each other reactor class can just be a single line.